### PR TITLE
Markup pages

### PR DESCRIPTION
**Pull request by Forsait**

### CODE REVIEW

File: src/components/error/ErrorBoundary.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+  static getDerivedStateFromError(error) {
+    // Update state so the next render will show the fallback UI.
+    return { hasError: true };
+  }
+
+  componentDidCatch(error, errorInfo) {
+    // You can also log the error to an error reporting service
+    // logErrorToMyService(error, errorInfo);
+  }
```
Comment (yanaSelin) [inline src/components/error/ErrorBoundary.js]: Please remove this redundant method

File: src/components/movie/MoviePage.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    const url = `/api/movies?offset=0&limit=6&sortBy=release_date&search=${genre}&searchBy=genres&sortOrder=desc`;
+
+    fetch(url)
+      .then(res => res.json())
+      .then((res) => {
+        console.log(res)
+        this.setState({
+          movieList: res.data,
+        });
+      })
```
Comment (yanaSelin) [inline src/components/movie/MoviePage.js]: Minor: Usually such request functionality is in service. Not a problem for this homework, but would be nice to move in future

File: src/components/movie/MoviePage.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  getReleaseYear(dateStr) {
+    if (!dateStr) return '';
+    return dateStr.slice(0, 4);
+  }
```
Comment (yanaSelin) [inline src/components/movie/MoviePage.js]: You use this in Movie also, so it can be moved to utils

File: src/components/movie/MoviePage.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    return <>
+      <div className={styles.container}>
+        <div className={styles.sub_header}>
+          <Brand />
+          <a href="#" className={styles.search_link}>
+            <img src={searchIcon} />
+          </a>
+        </div>
```
Comment (yanaSelin) [inline src/components/movie/MoviePage.js]: Header should be separated as component as it is used on both pages

File: src/components/movie/MoviePage.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+              </div>
+              <div className={styles.t4}>
+                <span className={styles.t4_1}>{this.state.movie.runtime || '-'}</span>
+                <span className={styles.t4_2}>min</span>
+              </div>
+            </div>
+            <div className={styles.description}>{this.state.movie.overview}</div>
+          </div>
+        </div>
+      </div>
```
Comment (yanaSelin) [inline src/components/movie/MoviePage.js]: This component it "smart" so it shouldn't contain any markup. This presentational logic should be extracted in separate presentational component (MoovieInfo for instance).

File: src/components/movie/MoviePage.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+      <div className={`${styles.hot} westside`}>
+        {genreView}
+      </div>
```
Comment (yanaSelin) [inline src/components/movie/MoviePage.js]: This component with different content but with same container stiles is used on both pages, so it should be separated.

For instance we can create simple component-wrapper with "westside" class and 2 additional presentational components that use it with different content.

File: src/components/search/Search.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+        </div>
+        <div className={`${styles.result_container} westside`}>
+          <div className={styles.q_1}>{resTotal}</div>
+          <Radiobuttons 
+            className={styles.r_1}
+            title={options.title}
+            list={options.list}
+            change={this.sortByChange}/>
+        </div>
+      </>
```
Comment (yanaSelin) [inline src/components/search/Search.js]: This component it "smart" so it shouldn't contain any markup. 
Scheme sample of decomposition:

```suggestion
      <Header />
      <Presentational component for form_container />
      <Presentational component for result_container /> - uses component-wrapper for corresponding component from MoviePage
```

File: webpack.config.js [hunk 8]
```diff
@@ -1,5 +1,8 @@
 const path = require('path');
-const HtmlWebPackPlugin = require("html-webpack-plugin");
+const HtmlWebPackPlugin = require('html-webpack-plugin');
+const { CleanWebpackPlugin } = require('clean-webpack-plugin');
+// const CopyPlugin = require('copy-webpack-plugin');
+// const ExtractTextPlugin = require('extract-text-webpack-plugin')
```
Comment (yanaSelin) [inline webpack.config.js:5]: Please remove redundant code

**Review by yanaSelin — COMMENTED** [review]
Using common structure will help will classifying and decomposition of components:

- components/ - presentational components
- containers/ - smart components (with business logic, connected to redux)
- pages/ - only 2 files in scope on this course, once we integrate redux they likely will not contain business logic or state and will just render smart components

File: src/components/search/Search.js (hunk 7 above)
Comment (Forsait) [reply src/components/search/Search.js]: <Presentational component for result_container /> also contain part of form "Sort by", so I don't want to wrap it in another component.

**Pull request merged**